contrib/initramfs/hooks/zfsunlock.in

```diff
@@ -0,0 +1,60 @@
+#!/bin/sh
+
+PREREQ="dropbear"
+
+prereqs() {
+    echo "$PREREQ"
+}
+
+case "$1" in
+    prereqs)
+        prereqs
+        exit 0
+    ;;
+esac
+
+. /usr/share/initramfs-tools/hook-functions
+
+if [ -r "${DESTDIR}/sbin/dropbear" ]; then
+  cat >"$DESTDIR"/bin/zfsunlock <<'EOF'
+#!/bin/sh
+if [ -e /run/zfsunlockcomplete ]; then
+  exit 0
+fi
+echo -n "Waiting for zfs init script to complete, press Ctrl-C to exit into shell ..."
+while [ ! -e /run/zfspassinpipe ] && [ ! -e /run/zfsunlockcomplete ]; do
+  sleep 1;
+  echo -n "."
+done
+if [ -e /run/zfsunlockcomplete ]; then
+  exit 0
+fi
+echo
+echo "Unlocking encrypted ZFS pools"
+echo "Enter correct password or press Ctrl-C to exit into shell"
+echo "After 3 incorrect passwords system will reboot"
```

Comment:
What is the point of rebooting after 3 attempts?

Reply:
Most simple countermeasure against guessing password attempts.

Reply:
I think a reasonable sleep would be more than sufficient.

Reply:
Doesn't the user have to authenticate to dropbear anyway? dropbear isn't a wide-open-to-the-Internet SSH server, is it?